code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -108,4 +108,16 @@
 	icon_state = "pipette"
 	amount_per_transfer_from_this = 1
 	possible_transfer_amounts = list(0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1)
-	volume = 1
\ No newline at end of file
+	volume = 1
+
+//Syndicate item. Virus transmitting mini hypospray
+/obj/item/reagent_containers/dropper/precision/viral_injector
+
+/obj/item/reagent_containers/dropper/precision/viral_injector/attack(mob/living/M, mob/living/user, def_zone)
+	if(M.can_inject(user, 1))
+		to_chat(user, "<span class='warning'>You stab [M] with the [src].</span>")
+		add_attack_logs(user, M, "Stabbed with Viral Injector")
+		
+		if(reagents.total_volume && M.reagents)

```

Comment:
I copied most of the logic from the sleepy pen. It currently also notifies you if you stabbed somebody even though the pen was empty.

Reply:
well that should probably be changed.

there is no reason to be informing people "you stab blah with the pen/viral injector" and have a log for it if NOTHING happens.

Reply:
REgardless of if it has stuff in it, you should still be able to stab people...

Reply:
Currently I've removed the log about the stabbing. Since yeah it's pretty useless. now it'll log the reagents and the viruses (with symptoms) injected if any.